src/pykoala/corrections/flux_calibration.py

```diff
@@ -239,105 +236,152 @@ def extract_stellar_flux(self, data_container,
             x, y = np.meshgrid(np.arange(data_container.n_cols),
                                np.arange(data_container.n_rows))
             x, y = x.flatten(), y.flatten()
+            x, y = data_container.wcs.celestial.pixel_to_world_values(x, y)
 
         data = data[wave_mask, :]
-        
+        variance = variance[wave_mask, :]
+
         # Declare variables
         residuals = []
         profile_popt = []
         profile_var = []
         running_wavelength = []
         raw_flux = []
+        cog = []
+        cog_var = []
         # Fitting tolerance and number of evaluations
         if 'ftol' not in fitter_args.keys():
             fitter_args['ftol'] = 0.001
             fitter_args['xtol'] = 0.001
             fitter_args['gtol'] = 0.001
         if 'max_nfev' not in fitter_args.keys():
-            fitter_args['max_nfev'] = 10000
+            fitter_args['max_nfev'] = 100000
  
         # Loop over all spectral slices
         self.corr_print("...Fitting wavelength chuncks...")
         for lambda_ in range(0, wavelength.size, wave_window):
             wave_slice = slice(lambda_, lambda_ + wave_window, 1)
-            mean_wave = np.nanmedian(wavelength[wave_slice])
+            wave_edges = [wavelength[wave_slice][0], wavelength[wave_slice][-1]]
             slice_data = data[wave_slice]
+            slice_var = variance[wave_slice]
             # Compute the median value only considering good values (i.e. excluding NaN)
             slice_data = np.nanmedian(slice_data, axis=0)
-            if not np.isfinite(slice_data).any():
-                self.corr_print("Chunk at {}+-{} contains no useful data"
-                                .format(mean_wave, wave_window))
+            slice_var = np.nanmedian(slice_var, axis=0) / np.sqrt(slice_var.shape[0])
+            slice_var[slice_var <= 0] = np.inf
+
+            if np.isfinite(slice_data).all():
+                self.corr_print("Chunk between {} to {} contains no useful data"
+                                .format(wave_edges[0], wave_edges[1]))
                 continue
-            # Positions without signal are set to 0.
-            slice_data = np.nan_to_num(slice_data, posinf=0., neginf=0.)
+            # Pixels without signal are set to 0.
+            mask = np.isfinite(slice_data) & np.isfinite(slice_var)
             ###################################################################
             # Computing the curve of growth
             ###################################################################
-            x0, y0 = centre_of_mass(slice_data, x, y)
-            y_min, y_max = y.min(), y.max()
-            x_min, x_max = x.min(), x.max()
+            x0, y0 = centre_of_mass(slice_data * mask, x, y)
             # Make the growth curve
-            r2, growth_c = growth_curve_1d(slice_data, x - x0, y - y0)
+            r2 = ((x - x0)**2 + (y - y0)**2) * 3600**2  # expressed in arcsec^2
+            r2_dummy = np.arange(0, 10**2, 0.5**2)
```

Comment:
I guess it's OK, but why 10 arcsec?
It also feels a bit hard-coded...

Reply:
I completely missed that. I has been removed and added as an argument in the method with the same default values.